connect/runtime/src/test/java/org/apache/kafka/connect/integration/TaskHandle.java

```diff
@@ -65,19 +57,36 @@ public void record() {
      * @param batchSize the number of messages
      */
     public void record(int batchSize) {
-        if (recordsRemainingLatch != null) {
-            IntStream.range(0, batchSize).forEach(i -> recordsRemainingLatch.countDown());
-        }
+        recordLatches.record(batchSize);
         connectorHandle.record(batchSize);
     }
 
+    /**
+     * Record a message arrival at the task and the connector overall.
+     *
+     * @param topic the name of the topic
+     */
+    public void record(String topic) {
+        recordLatches.record(topic);
+        connectorHandle.record(topic);
+    }
+
+    /**
+     * Record arrival of a batch of messages at the task and the connector overall.
+     *
+     * @param topic     the name of the topic
+     * @param batchSize the number of messages
+     */
+    public void record(String topic, int batchSize) {
+        recordLatches.record(topic, batchSize);
+        connectorHandle.record(topic, batchSize);
+    }
+
```

Comment:
Step 5: Added logic to record (and also expect, see below) number of records that a task has seen for a given topic.